### DISCUSSION TITLE
CCSDS in Fprime-GDS

### DISCUSSION THREAD

**Discussion by ebenezerarunkumar** [nasa/fprime#3026] [Q&A] [answered]
Has anyone extended the fprime-gds to accept CCSDS wrapped telemetry & events and send out CCSDS wrapped commands? Would be interested in looking at an existing implementation if there is one!

**Comment by ReggieMarr** [comment]
I don't think fprime supports CCSDS writ large. That being said [this MR](https://github.com/nasa/fprime/pull/2900) provided some refactoring to the communications pipeline to as part of an effort to make this happen. There also is [this MR](https://github.com/nasa/fprime-gds/pull/179) which appears to add some support on the fprime-gds side.

I created a fork based off the former branch, rebased it onto the devel branch and have been adding datalink support for CCSDS, using [this reference project](https://github.com/ReggieMarr/space-vehicle-system/tree/using_ccsds_framers_deframers) for testing. While I also use fprime-gds for some testing I'm more of a [yamcs[(https://yamcs.org/) guy.

That being said I plan on using the hub model to handle the ccsds communications link between the space vehicle and ground and then can easily use a regular tcp/udp client/server to interact with linked data post-deframing into classic fprime like frames thereby bypassing the need for fprime-gds to support handling the ccsds link itself.

I'm curious if that sounds like something that would work for you and if not why.

Also which CCSDS protocol(s) are you looking at and how are you envisioning mapping fprime events/telemetry/commands onto that ? I've currently been playing with a concept that maps commands/command responses and events to TC Space Data Link and Telemetry to the TM Space Data Link but I know that SPP is also a popular choice for similar use cases.

**Comment by LeStarch** [comment, marked as answer]
We are now working on supporting CCSDS in F Prime.  It is scheduled for the next release

**Comment by thomas-bc** [comment]
Annndddd.... merged in devel! 
https://github.com/nasa/fprime/pull/3684

Wohooo! 🎉

**Reply by zimri-leisher** [reply]
yay!

**Reply by ebenezerarunkumar** [reply]
amazing!
